### PR TITLE
[HACK week] Deposits summary view

### PR DESCRIPTION
**Pull request by joshheald**
Merge after: #10821 

## Description

In this HACK week project, we're adding the Deposit Overview, so merchants can see their WooPayments balance, and the money currently on its way to their account.

This PR adds the view for the deposits. It's feature flagged until it can be polished for production, but this is planned in a future project. Merging it now will help us avoid issues when we come to finish it off.

Apologies for the length – I could split out the `TopTabView` if that would be helpful?

## Testing instructions

Unit tests will follow in a future PR

Using a store with WooPayments, ideally multi-currency enabled, and some test transactions:
1. Navigate to `Menu > Payments > WooPayments Balances`
2. Observe that the relevant amounts match those shown on the web for your store at `wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview`

Note that only the most recent deposit is shown in the app; the web shows a few recent deposits.

Using a non-WooPayments store
1. Navigate to `Menu > Payments`
2. Observe that the WooPayments Balances item is not shown

Using a WooPayments store again, set up Xcode to build a `Release` scheme to a simulator
1. Navigate to `Menu > Payments`
2. Observe that the WooPayments Balances item is not shown

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/2472348/ebf0e9c3-5e1f-493f-9094-d78c1542cdda

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Model/WooPaymentsDepositsOverview.swift [hunk 1]
```diff
@@ -109,13 +136,124 @@ public struct WooPaymentsDepositsSchedule: Codable, GeneratedFakeable, Generated
...
+            guard let dayNames = dateFormatter.standaloneWeekdaySymbols,
+                  let localizedDayName = dayNames[safe: dayIndex] else {
+                // This shouldn't really happen... but at least it'll be sort of understandable.
+                return rawValue
+            }
+
+            return localizedDayName
+        }
+
+        private var dayIndex: Int {
```
Comment (toupper) [inline Networking/Networking/Model/WooPaymentsDepositsOverview.swift:244]: Perhaps we can use `allCases` for this index? We could then use `firstIndex(where:)`, as by default, the order of he synthesized allCases is the same order as their declaration.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift [hunk 2]
```diff
@@ -211,9 +211,17 @@ private extension InPersonPaymentsMenuViewController {
     ///     If not set, this function will fetch the value from the viewModel, however, when called in response to a @Published update,
     ///     the viewModel value may not yet be up to date.
     func configureSections(isEligibleForTapToPayOnIPhone: Bool? = nil,
-                           shouldShowTapToPayOnIPhoneFeedback: Bool? = nil) {
+                           shouldShowTapToPayOnIPhoneFeedback: Bool? = nil,
+                           depositsOverviewViewModels: [WooPaymentsDepositsCurrencyOverviewViewModel]? = nil) {
         var composingSections: [Section?] = [actionsSection]
 
+        let depositsOverviewViewModels = depositsOverviewViewModels ?? viewModel.depositsOverviewViewModels
+        if #available(iOS 16.0, *),
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift:219]: Nice one just focusing on iOS 16 onwards

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift [hunk 3]
```diff
@@ -259,6 +267,10 @@ private extension InPersonPaymentsMenuViewController {
         return Section(header: Localization.paymentOptionsSectionTitle, rows: [.managePaymentGateways])
     }
 
+    func depositsSection(from viewModels: [WooPaymentsDepositsCurrencyOverviewViewModel])-> Section {
+        return Section(header: "Woo Payments Balance", rows: [Row.depositOverview])
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift]: Do we need to localize that header? `"Woo Payments Balance"`

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift [hunk 4]
```diff
@@ -391,6 +405,12 @@ private extension InPersonPaymentsMenuViewController {
         cell.accessoryType = .none
     }
 
+    func configureDepositOverviews(cell: LeftImageTableViewCell) {
+        prepareForReuse(cell)
+        cell.accessibilityIdentifier = "deposit-overviews"
+        cell.configure(image: .bankIcon, text: "WooPayments Deposits")
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift]: Same here about string localization

**Review by toupper — APPROVED** [review]
It tests well and code looks good as well, it's a nice addition for sure! 🚢 :shipit:

File: Networking/Networking/Model/WooPaymentsDepositsOverview.swift (hunk 1 above)
Comment (joshheald) [reply Networking/Networking/Model/WooPaymentsDepositsOverview.swift:244]: I did think about that, but it seemed like one of those places where explicit is better – if we used the synthesised default, it only needs someone to switch the enum to be Monday-first and it'll break... in a way we're unlikely to notice.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift (hunk 4 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift]: Thanks! Both done in 04902ea

**Pull request merged**